### PR TITLE
Low node utilization test cases.

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
/cc @aveshagarwal test case for low node utilization strategy.

### CODE REVIEW

File: pkg/rescheduler/strategies/lownodeutilization.go [hunk 1]
```diff
@@ -43,12 +43,16 @@ func LowNodeUtilization(client clientset.Interface, strategy api.ReschedulerStra
 	if !strategy.Enabled {
 		return
 	}
+	evictPodsFromOverUtilizedNodes(client, strategy, evictionPolicyGroupVersion, nodes)
```
Comment (aveshagarwal) [inline pkg/rescheduler/strategies/lownodeutilization.go]: I am thinking, if you could rename it to evictPodsFromTargetedNodes.

File: pkg/rescheduler/strategies/lownodeutilization.go [hunk 2]
```diff
@@ -90,16 +95,16 @@ func LowNodeUtilization(client clientset.Interface, strategy api.ReschedulerStra
...
 	} else if len(lowNodes) == len(nodes) {
 		fmt.Printf("all nodes are underutilized\n")
-		return
+		return 0
 	} else if len(targetNodes) == 0 {
 		fmt.Printf("no node is above target utilization\n")
-		return
+		return 0
 	}
 	SortNodesByUsage(targetNodes)
```
Comment (aveshagarwal) [inline pkg/rescheduler/strategies/lownodeutilization.go]: Also I think move everything above in the original function. And move stuff below in the new function.

File: pkg/rescheduler/strategies/lownodeutilization.go [hunk 3]
```diff
@@ -216,15 +224,14 @@ func NodeUtilization(node *v1.Node, pods []*v1.Pod) (api.ResourceThresholds, []*
...
-
 	totalReqs := map[v1.ResourceName]resource.Quantity{}
 	for _, pod := range pods {
 		sr, err := podutil.CreatorRef(pod)
 		if err != nil {
 			sr = nil
 		}
 
-		if podutil.IsMirrorPod(pod) || podutil.IsPodWithLocalStorage(pod) || sr == nil || podutil.IsDaemonsetPod(sr) {
+		if podutil.IsMirrorPod(pod) || podutil.IsPodWithLocalStorage(pod) || sr == nil || podutil.IsDaemonsetPod(sr) || podutil.IsCriticalPod(pod) {
```
Comment (aveshagarwal) [inline pkg/rescheduler/strategies/lownodeutilization.go]: I am still wondering why this is happening.

File: pkg/rescheduler/strategies/lownodeutilization.go (hunk 1 above)
Comment (ravisantoshgudimetla) [reply pkg/rescheduler/strategies/lownodeutilization.go]: ack.

File: pkg/rescheduler/strategies/lownodeutilization.go (hunk 2 above)
Comment (ravisantoshgudimetla) [reply pkg/rescheduler/strategies/lownodeutilization.go]: Well, I actually thought about that but 
- We have the pass just the values like lowNodes,TargetNodes etc to the evictPodsFromOverUtilized function not client etc that we will be building as part of unit tests.
- Just to be consistent with other strategy where we have a registration function and it delegates to other function which does everything else.

File: pkg/rescheduler/strategies/lownodeutilization.go (hunk 3 above)
Comment (ravisantoshgudimetla) [reply pkg/rescheduler/strategies/lownodeutilization.go]: I am confused too.

File: pkg/rescheduler/strategies/lownodeutilization.go [hunk 4]
```diff
@@ -43,37 +43,69 @@ func LowNodeUtilization(client clientset.Interface, strategy api.ReschedulerStra
...
 	if targetThresholds == nil {
 		fmt.Printf("no target resource threshold is configured\n")
-		return
+		return false
 	} else if _, ok := targetThresholds[v1.ResourcePods]; !ok {
 		fmt.Printf("no target resource threshold for pods is configured\n")
-		return
+		return false
 	}
+	return false
```
Comment (aveshagarwal) [inline pkg/rescheduler/strategies/lownodeutilization.go]: return true.

Comment (ravisantoshgudimetla) [reply pkg/rescheduler/strategies/lownodeutilization.go]: Oops. Copy paste issue. Changed it.

**Comment by aveshagarwal** [comment]
@ravisantoshgudimetla  see this error, its failing to merge in lownodeutilization.go.rej:

```
patching file pkg/rescheduler/strategies/duplicates_test.go
patching file pkg/rescheduler/strategies/lownodeutilization.go
Hunk #8 FAILED at 231.
1 out of 8 hunks FAILED -- saving rejects to file pkg/rescheduler/strategies/lownodeutilization.go.rej
patching file pkg/rescheduler/strategies/lownodeutilization_test.go
patching file pkg/rescheduler/strategies/lownodeutilization.go
patching file pkg/rescheduler/strategies/lownodeutilization_test.go
patching file pkg/rescheduler/strategies/lownodeutilization.go
```

And the error is due to:

```
--- pkg/rescheduler/strategies/lownodeutilization.go
+++ pkg/rescheduler/strategies/lownodeutilization.go
@@ -231,7 +238,7 @@
 			sr = nil
 		}
 
-		if podutil.IsMirrorPod(pod) || podutil.IsPodWithLocalStorage(pod) || sr == nil || podutil.IsDaemonsetPod(sr) {
+		if podutil.IsMirrorPod(pod) || podutil.IsPodWithLocalStorage(pod) || sr == nil || podutil.IsDaemonsetPod(sr) || podutil.IsCriticalPod(pod) {
 			nonRemovablePods = append(nonRemovablePods, pod)
 			if podutil.IsBestEffortPod(pod) {
 				continue
``` 
Can you squash and rebase and resubmit? I am not sure where is the issue github or your repo?

**Pull request merged**